docker/docker-compose.yml

```diff
@@ -1,10 +1,22 @@
 version: '3.4'
 
 services:
-  web:
+  app:
```

Comment:
do you think we can change the name from `app` to `search` or `libero-search` and `web` to `nginx`?

Reply:
Looking at https://github.com/libero/browser/blob/master/docker-compose.yaml we tend to go for role more than for implementation, for example you could replace Nginx with Apache (hypothetical) without the service name changing.
In some cases I renamed `app` to `wsgi` which is the actual role (listening for connections using the WSGI-protocol), but the service `app` remains useful for `docker-compose run app python search/something.py` so it's not true that it's only WSGI-related if it can be used for command line processes.

https://github.com/libero/search/pull/8/commits/dcad5e75c3099282f137be239bb6efe917a55fb5 now forces explanatory container names such as `search_app_1`.

Reply:
I like this change. Does docker know to look for `.env` automagically?

Reply:
[`docker-compose` does](https://docs.docker.com/compose/env-file/): the values are either used directly by it or available for interpolation in `docker-compose.yml` files.

Reply:
I ask because there is a .env file in the `docker/` directory. Can we repurpose this?

Reply:
Ideally, I don't want files in the root directory if it can be avoided.

Reply:
They are not related apart from the extension; `docker-compose` won't read `dev.env` automatically (because of the name) and it is [explicitly referenced](https://github.com/libero/search/blob/master/docker/docker-compose.dev.yml#L17) for its variables to be passed to the `app` service containers. You don't want `COMPOSE_PROJECT_NAME` to be passed to the Python container (it doesn't need it) and at the same time you don't need `FLASK_ENV` to be available for interpolation in `docker-compose.yml`.

Reply:
So the `.env` file needs to be in the root directory or can it live in the same directory as the docker-compose file?

Reply:
> Ideally, I don't want files in the root directory if it can be avoided.

Unfortunately the documentation (just linked) doesn't provide for a way to customize the path to it, and we would have to execute `docker-compose` exclusively from inside the `docker/` directory which would be annoying (and error-prone as it won't produce an error if run from the root, just ignore a few configurations).

Reply:
yeah that's not ideal. Leave it where it is if that's the intended design.

Reply:
(since it's needed for rendering the `docker-compose.yml`, it's probably read before that and hence there's no place to set a configuration for its path)